Remove deprecated "Errors" types

jikim-msft opened the pull request:
#### Description

Related PR: https://github.com/microsoft/FluidFramework/pull/17651

This PR applies the changes merged in the `next` branch to the `main` branch.

jikim-msft committed:
fix errors

jikim-msft committed:
Merge branch 'microsoft:main' into main

jikim-msft commented on experimental/dds/ot/ot/README.md:
```diff
@@ -32,7 +32,7 @@ npm i @fluid-experimental/ot
 ## API Documentation
 
-API documentation for **@fluid-experimental/ot** is available at <https://fluidframework.com/docs/api/v1/ot>.
+API documentation for **@fluid-experimental/ot** is available at <https://fluidframework.com/docs/apis/ot>.
```
[Build 227406](https://dev.azure.com/fluidframework/public/_build/results?buildId=227406&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=3f6c1bb4-e2b1-5180-b7f3-9a06f477b8ee&l=14)

@Josmithr 

Pipeline is complaining about extraneous modified files `api-docs-link-section.md` and `library-package-readme.md`, but I think I should also revert the changes (`api/v1`  back to `/apis`) in other `README` files?

Josmithr replied:
@chentong7 is currently working on resolving these. I would recommend just reverting these changes on your PR for now.

jikim-msft committed:
change doc

jikim-msft committed:
fix api doc

jikim-msft committed:
fix doc

jikim-msft committed:
fix api-docs-link-section

jikim-msft committed:
fix doc

jikim-msft committed:
rever api docs

jikim-msft committed:
fix release tag from alpha to public

jikim-msft committed:
Remove deprecated "Errors" types on next-branch (#17651)

[5467](https://dev.azure.com/fluidframework/internal/_workitems/edit/5467/)

Deprecate error types on the next branch.

---------

Co-authored-by: Joshua Smithrud <54606601+Josmithr@users.noreply.github.com>
Co-authored-by: Alex Villarreal <716334+alexvy86@users.noreply.github.com>

jikim-msft committed:
remove space

jikim-msft committed:
fix tags

jikim-msft committed:
Merge branch 'microsoft:main' into main

tylerbutler commented on .changeset/cold-deer-cheat.md:
Do you want both of these changesets? they look very similar to one another - maybe a mistake?

jikim-msft replied:
You're right. I deleted the file.

tylerbutler commented on .changeset/cold-deer-cheat.md:
```diff
@@ -0,0 +1,20 @@
+---
+"@fluidframework/container-definitions": major
+"@fluidframework/container-loader": major
+"@fluidframework/container-runtime": major
+"@fluidframework/datastore": major
+"@fluidframework/driver-definitions": major
+"@fluidframework/driver-utils": major
+"fluid-framework": major
+"@fluidframework/odsp-doclib-utils": major
+"@fluidframework/odsp-driver": major
+"@fluidframework/odsp-driver-definitions": major
+"@fluidframework/routerlicious-driver": major
+"@fluidframework/telemetry-utils": major
+"@fluid-private/test-end-to-end-tests": major
+"@fluidframework/tinylicious-client": major
```
Please change all of these to "minor" instead of major. This is just a side effect because the change originated on the next branch.

jikim-msft replied:
Changed to `minor`.

tylerbutler commented on .changeset/mean-mugs-suffer.md:
```diff
@@ -0,0 +1,20 @@
+
+Removes a number of deprecated error related enums, and replace them with new object-based based enumerations.
+
+The changes involve removing deprecated error related enums ContainerErrorType, DriverErrorType, OdspErrorType and RouterliciousErrorType and replacement to new object-based enumerations of ContainerErrorTypes, DriverErrorTypes, OdspErrorTypes and RouterliciousErrorTypes. Refer to release notes of version [Fluid Framework v2.0.0-internal.7.0.0](https://github.com/microsoft/FluidFramework/releases ...
```
```suggestion
The changes involve removing deprecated error related enums ContainerErrorType, DriverErrorType, OdspErrorType and RouterliciousErrorType and replacement to new object-based enumerations of ContainerErrorTypes, DriverErrorTypes, OdspErrorTypes and RouterliciousErrorTypes. Refer to release notes of version [Fluid Framework v2.0.0-internal.7.0.0](https://github.com/microsoft/FluidFramework/releases/tag/client_v2.0.0-internal.7.0.0).
```

jikim-msft committed:
Update .changeset/mean-mugs-suffer.md

Co-authored-by: Tyler Butler <tyler@tylerbutler.com>

jikim-msft committed:
delete changeset / change to minor

jikim-msft committed:
Merge branch 'main' of https://github.com/jikim-msft/FluidFramework

jikim-msft committed:
Merge branch 'microsoft:main' into main

Josmithr commented on packages/drivers/routerlicious-driver/src/test/errorUtils.spec.ts:
```diff
@@ -4,67 +4,79 @@
 describe("ErrorUtils", () => {
+	/**
+	 * Checks if the input is an {@link IThrottlingWarning}.
+	 * */
```
```suggestion
	 */
```

jikim-msft committed:
Update packages/drivers/routerlicious-driver/src/test/errorUtils.spec.ts

Co-authored-by: Joshua Smithrud <54606601+Josmithr@users.noreply.github.com>
